### PR TITLE
Can deploy spring boot onto web app

### PR DESCRIPTION
**Pull request by jdneo**
Can deploy spring boot onto web app

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.core/src/com/microsoft/azuretools/core/utils/MavenUtils.java [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+            return build.getDirectory() + File.separator + build.getFinalName() + "." + mavenProject.getPackaging();
+        }
+        return "";
+    }
+
+    public static void executePackageGoal(@NotNull IProject project) throws Exception {
+        final IFile pom = getPomFile(project);
+        final List<String> goals = Arrays.asList(MAVEN_CLEAN, MAVEN_PACKAGE);
+        final IMaven maven = MavenPlugin.getMaven();
+        if (maven == null) {
```
Comment (adashen) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.core/src/com/microsoft/azuretools/core/utils/MavenUtils.java]: throw exception in this case?

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.core/src/com/microsoft/azuretools/core/utils/MavenUtils.java [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        MavenExecutionResult result = context.execute(new ICallable<MavenExecutionResult>() {
+            @Override
+            public MavenExecutionResult call(IMavenExecutionContext context, IProgressMonitor innerMonitor) throws CoreException {
+             return ((MavenImpl)maven).lookupComponent(Maven.class)
+                   .execute(request);
+            }
+          }, monitor);
+        List<Throwable> exceptions = result.getExceptions();
+        if (exceptions.size() > 0) {
+            throw new Exception(exceptions.get(0));
```
Comment (adashen) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.core/src/com/microsoft/azuretools/core/utils/MavenUtils.java]: group the exceptions?

**Comment by jdneo** [comment]
retest this please.

**Comment by Eskibear** [comment]
retest this please

**Comment by Eskibear** [comment]
retest this please

**Comment by adashen** [comment]
retest this please

**Comment by adashen** [comment]
retest this please

**Pull request merged**